cli/ballerina-launcher/src/main/java/org/ballerinalang/launcher/LauncherUtils.java

```diff
@@ -118,7 +117,7 @@ public static void runProgram(Path sourceRootPath, Path sourcePath, boolean runS
         } else {
             runMain(programFile, args);
         }
-
+        BLangProgramRunner.resumeStates(programFile);
```

Comment:
Shouldn't we have done this for the main function as well?

Reply:
The resume is not depended on either service or main. Hence if there are any states which are persisted, will be resumed after running service or main. Note: currently interruptible only works with services.